### ISSUE TITLE
BUG: Typed local variable in block raises nil assertion

### ISSUE THREAD

**Issue by aljelly** [crystal-lang/crystal#7037]
Code:
```crystal
1.times do
  time : Time::Span
end

# or

loop do
  time : Time::Span
  break
end
```

Seems to only happen within blocks (doesn't happen in while loops).

Result:
```
$ ./bugtest
Nil assertion failed at /crystal/src/compiler/crystal/semantic/bindings.cr:3 (Exception)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Error: you've found a bug in the Crystal compiler. Please open an issue, including source code that will allow us to reproduce the bug: https://github.com/crystal-lang/crystal/issues
```

Crystal Version:
```
$ crystal -v
Crystal 0.27.0 [c9d1eef8f] (2018-11-01)

LLVM: 4.0.0
Default target: x86_64-unknown-linux-gnu
```

**Comment by straight-shoota** [comment]
Debug backtrace:
```
Nil assertion failed (NilAssertionError)
  from /home/vagrant/src/nil.cr:106:5 in 'not_nil!'
  from /home/vagrant/src/compiler/crystal/semantic/bindings.cr:0:5 in 'dependencies'
  from /home/vagrant/src/compiler/crystal/codegen/codegen.cr:266:5 in 'bound_to_mod_nil?'
  from /home/vagrant/src/compiler/crystal/codegen/codegen.cr:1769:36 in 'reset_block_vars'
  from /home/vagrant/src/compiler/crystal/codegen/codegen.cr:1461:11 in 'visit'
  from /home//vagrant/src/compiler/crystal/syntax/visitor.cr:27:12 in 'accept'
  from /home/vagrant/src/compiler/crystal/codegen/codegen.cr:2074:7 in 'accept'
  from /home/vagrant/src/compiler/crystal/codegen/codegen.cr:804:11 in 'visit'
  from /home/vagrant/src/compiler/crystal/syntax/visitor.cr:27:12 in 'accept'
  from /home/vagrant/src/compiler/crystal/codegen/codegen.cr:2074:7 in 'accept'
  from /home/vagrant/src/compiler/crystal/codegen/call.cr:286:13 in 'codegen_call_with_block'
  from /home/vagrant/src/compiler/crystal/codegen/call.cr:32:9 in 'visit'
  from /home/vagrant/src/compiler/crystal/syntax/visitor.cr:27:12 in 'accept'
  from /home/vagrant/src/compiler/crystal/codegen/codegen.cr:2074:7 in 'accept'
  from /home/vagrant/src/compiler/crystal/codegen/codegen.cr:605:9 in 'visit'
  from /home/vagrant/src/compiler/crystal/syntax/visitor.cr:27:12 in 'accept'
  from /home/vagrant/src/compiler/crystal/codegen/codegen.cr:2074:7 in 'accept'
  from /home/vagrant/src/compiler/crystal/codegen/codegen.cr:65:7 in 'codegen'
  from /home/vagrant/src/compiler/crystal/codegen/codegen.cr:63:5 in 'codegen:debug:single_module'
  from /home/vagrant/src/compiler/crystal/compiler.cr:22:7 in 'codegen'
  from /home/vagrant/src/compiler/crystal/compiler.cr:153:16 in 'compile'
```

**Comment by caspiano** [comment]
Not reproducible as of Crystal 1.6.0

Resolved by #12405

https://carc.in/#/r/e020
